### PR TITLE
KfConfig changes

### PR DESCRIPTION
**Pull request by lluunn**
Port from #4250, only the kfconfig part.

/cc @gabrielwen @jlewi

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4295)

### CODE REVIEW

File: bootstrap/pkg/apis/apps/configconverters/converters.go [hunk 1]
```diff
@@ -123,7 +125,24 @@ func LoadConfigFromURI(configFile string) (*kfconfig.KfConfig, error) {
...
+		return nil, &kfapis.KfError{
+			Code:    int(kfapis.INTERNAL_ERROR),
+			Message: fmt.Sprintf("could not get current directory for KfDef %v", err),
+		}
+	}
+
+	return converter.ToKfConfig(cwd, configFileBytes)
+}
+
+func isCwdEmpty() string {
```
Comment (gabrielwen) [inline bootstrap/pkg/apis/apps/configconverters/converters.go]: where is this func used?

File: bootstrap/pkg/apis/apps/kfconfig/types.go [hunk 2]
```diff
@@ -368,3 +599,71 @@ func IsConditionNotFound(e error) bool {
...
+	}
+}
+
+func IsSecretNotFound(e error) bool {
+	if e == nil {
+		return false
+	}
+	_, ok := e.(*SecretNotFound)
+	return ok
+}
```
Comment (gabrielwen) [inline bootstrap/pkg/apis/apps/kfconfig/types.go]: are we still using this?  should we migrate to `KfError`?

Comment (lluunn) [reply bootstrap/pkg/apis/apps/kfconfig/types.go]: discussed offline. let's migrate them later

File: bootstrap/pkg/apis/apps/configconverters/converters.go (hunk 1 above)
Comment (lluunn) [reply bootstrap/pkg/apis/apps/configconverters/converters.go]: removed. I think it's moved to somewhere else.

**Comment by gabrielwen** [comment]
have we reached the caps for some resources?
```
Creating kfctl-d9fd error: &{Code:RESOURCE_ERROR Location:/deployments/kfctl-d9fd/resources/kf
ctl-d9fd Message:{\"ResourceType\":\"gcp-types/container-v1beta1:projects.locations.clusters\",\"ResourceErrorCode\":\"FAILED_OPERATION\",\"ResourceErrorMessage\":\"The network \\
\"default\\\" does not have available private IP space in 10.0.0.0/8 to reserve a /14 block for containers for cluster {us-central1-a, 29647740582, kubeflow-ci-deployment, kfctl-d
9fd, 09ff39955a3859f8d9bcef06ce75a65b116e6dd929906aafc1d25b8b0a219b1d}. - https://container.googleapis.com/v1beta1/projects/29647740582/zones/us-central1-a/clusters/kfctl-d9fd\"}
```

**Comment by gabrielwen** [comment]
> have we reached the caps for some resources?
> 
> ```
> Creating kfctl-d9fd error: &{Code:RESOURCE_ERROR Location:/deployments/kfctl-d9fd/resources/kf
> ctl-d9fd Message:{\"ResourceType\":\"gcp-types/container-v1beta1:projects.locations.clusters\",\"ResourceErrorCode\":\"FAILED_OPERATION\",\"ResourceErrorMessage\":\"The network \\
> \"default\\\" does not have available private IP space in 10.0.0.0/8 to reserve a /14 block for containers for cluster {us-central1-a, 29647740582, kubeflow-ci-deployment, kfctl-d
> 9fd, 09ff39955a3859f8d9bcef06ce75a65b116e6dd929906aafc1d25b8b0a219b1d}. - https://container.googleapis.com/v1beta1/projects/29647740582/zones/us-central1-a/clusters/kfctl-d9fd\"}
> ```

looks like it's transient.

**Comment by lluunn** [comment]
still seeing
```
{Code:RESOURCE_ERROR Location:/deployments/kfc
tl-6d9c/resources/kfctl-6d9c Message:{\"ResourceType\":\"gcp-types/container-v1beta1:projects.locations.clusters\",\"ResourceErrorCode\":\"FAILED_OPERATION\",\
"ResourceErrorMessage\":\"The network \\\"default\\\" does not have available private IP space in 10.0.0.0/8 to reserve a /14 block for containers for cluster
{us-central1-a, 29647740582, kubeflow-ci-deployment, kfctl-6d9c, 95fcdcc24f34b1d6da4f3c9180e642571bbb34b960fb9d56d9e14597dee0a20b}. - https://container.googlea
pis.com/v1beta1/projects/29647740582/zones/us-central1-a/clusters/kfctl-6d9c\"} ForceSendFields:[] NullFields:[]}
```

**Comment by gabrielwen** [comment]
I saw the test passed that point before lunch..

**Comment by lluunn** [comment]
https://github.com/kubeflow/testing/issues/480

**Comment by lluunn** [comment]
Seems all workflows are green, not sure why it shows failure...

**Comment by lluunn** [comment]
failure was metadata deployment not ready in basic auth.

/retest

**Pull request merged**